### PR TITLE
Add real widget test using StatefulBuilder

### PR DESCRIPTION
**Pull request by gamebox**
This could easily be expanded upon, but currently replicates the behavior
that we attempted to test in the extant test case that was marked as `skip`

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -29,12 +29,9 @@ dependencies:
...
 dev_dependencies:
   build_runner: ^1.0.0
-  built_value_generator:
-    git: 
-      url: https://github.com/efortuna/built_value.dart.git/
-      path: built_value_generator
+  built_value_generator: any
 
-  test: ^1.5.1
+  test: 1.6.2
```
Comment (filiph) [inline pubspec.yaml:34]: This needs to be pinned like that? Just checking.

File: test/widget_test.dart [hunk 2]
```diff
@@ -7,53 +7,55 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_test/flutter_test.dart';
 import 'package:hn_app/main.dart';
+import 'package:hn_app/src/widgets/headline.dart';
```
Comment (filiph) [inline test/widget_test.dart:9]: This line is just below, commented out. Can you remove the commented line?

**Review by filiph — APPROVED** [review]
LGTM with 2 comments. Thanks!

File: pubspec.yaml (hunk 1 above)
Comment (gamebox) [reply pubspec.yaml:34]: It was the only way I could get it to build.  Maybe we could unpin when we have more time to update all of the deps?

**Comment by filiph** [comment]
Thanks!

**Pull request merged**